givitsite/gatherer/tasks.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import unicode_literals
+from background_task import background
+from friendreq.models import ItemRequest
+from friendreq.models import ItemsFound
+from .agora import givit_main
+
+
+@background(schedule=60)
+def gatherer_task():
+    iseek_dict = {'20016': 'bed', '20009': 'closet', '20008': 'chair',
+                  '20017': 'sofa', '10006': 'fridge', '10029': 'Washing machine'}
+    item_list = givit_main()
+    counter = 0
```

Comment:
Since `agora.py` already imports and uses model objects, I'd expect this final conversion from results to model to be there rather then here. But generally I'd expect most data conversion logic to reside in the model classes themselves.

In fact, couldn't `givit_main` be a background task on its own instead of needing this wrapper function?

Reply:
i red that the background-task could only be in tasks.py am i wrong? you mean to take givit_main and put it in tasks.py?

Reply:
Well, looking at the docs, it seems like you could use the `@background` decorator anywhere.

I did find this:
> To get the functions decorated by background picked up by the auto discovery mechanism, they must be placed in a file named tasks.py in your module, eg. myapp/tasks.py 

But I'm not sure what is this "auto discovery" they are talking about, I can't find it mentioned anywhere else in the docs. I'd try to simply place the decorator over the relevant functions and see if it works (I also remember seeing an example somewhere of a task function that is placed in a view file).

In any case you should put all the related functionality in one place and not split it arbitrarily.